File: src_python/equipamento.py
class Equipamento:
    def __init__(self, tag, descricao, ativo):
        self._tag = tag
        self._descricao = descricao
        self._ativo = ativo

    def ligar(self):
        self._ativo = True

    def desligar(self):
        self._ativo = False

    @property
    def tag(self):
        return self._tag

    def exibir_resumo(self):
        print(f"[Equipamento] {self._tag} - {self._descricao} - ativo={self._ativo}")

File: src_cpp/equipamento.cpp
#include "equipamento.hpp"

#include <iostream>

Equipamento::Equipamento(const std::string& tagEquipamento, const std::string& descricaoEquipamento, bool ativoInicial)
    : tag(tagEquipamento), descricao(descricaoEquipamento), ativo(ativoInicial) {}

void Equipamento::ligar() {
    ativo = true;
}

void Equipamento::desligar() {
    ativo = false;
}

std::string Equipamento::getTag() const {
    return tag;
}

bool Equipamento::estaAtivo() const {
    return ativo;
}

void Equipamento::exibirResumo() const {
    std::cout << "[Equipamento] " << tag << " - " << descricao
              << " - ativo=" << (ativo ? "sim" : "nao") << "\n";
}
